pkg/auth/dependency/interaction/provider_action.go

```diff
@@ -152,7 +156,19 @@ func (p *Provider) setupAuthenticator(i *Interaction, step *StepState, astate *m
 		return nil, ErrInvalidAction
 	}
 
-	// TODO(interaction): special handling for OTP
+	switch as.Type {
+	case AuthenticatorTypePassword:
+		// Nothing special needs to be done
+		break
+	case AuthenticatorTypeOOBOTP:
+		_, err := p.Authenticator.Authenticate( /* userID */ "", as, astate, secret)
```

Comment:
`i.UserID` should be passed in as userID.

Reply:
The user ID actually is unused because OOB OTP does not look at it. I will change it though.